packages/push/src/push_vm/stack.rs

```diff
@@ -212,22 +212,19 @@ impl<T> Stack<T> {
     /// Returns `StackError::Underflow` error if the stack has less than
     /// two elements.
     pub fn top2(&self) -> Result<(&T, &T), StackError> {
-        if self.size() >= 2 {
-            let x = self.top()?;
-            let y = self
-                .values
-                .get(self.size() - 2)
-                .ok_or(StackError::Underflow {
-                    num_requested: 2,
-                    num_present: 1,
-                })?;
-            Ok((x, y))
-        } else {
-            Err(StackError::Underflow {
+        let index_second_to_top = self.size().checked_sub(2).ok_or(StackError::Underflow {
+            num_requested: 2,
+            num_present: self.size(),
+        })?;
+        let x = self.top()?;
+        let y = self
+            .values
+            .get(index_second_to_top)
```

Comment:
We end up essentially checking the that there are at least two elements twice here, right? (Once here and once on line 215.) I doubt it matters a whole lot, but I wanted to make sure.

Reply:
yes, but we did the same thing before with the if check - this .get() here should now never be able to fail, just the same as it was before